internal/service/flexcluster/resource.go

```diff
@@ -196,3 +205,27 @@ func splitFlexClusterImportID(id string) (projectID, clusterName *string, err er
 
 	return
 }
+
+func isUpdateAllowed(ctx context.Context, plan, state *TFModel) (bool, error) {
+	if !plan.ProjectId.Equal(state.ProjectId) {
```

Comment:
curious why are we adding this validation here and not letting the API fail?

Reply:
API would fail with errors like `GROUP_NOT_FOUND`, `CLUSTER_NOT_FOUND` because we would be trying to update a non-existing cluster or a cluster in a non-existing project, which I think can be confusing for the user